domain-management/src/main/java/org/jboss/as/domain/management/ConfigurationChangeResourceDefinition.java

```diff
@@ -0,0 +1,241 @@
+
+    public static final SimpleAttributeDefinition MAX_HISTORY = SimpleAttributeDefinitionBuilder.create(
+            ModelDescriptionConstants.MAX_HISTORY, ModelType.INT)
+            .setDefaultValue(new ModelNode(10))
+            .build();
+    public static final PathElement PATH = PathElement.pathElement(SERVICE, CONFIGURATION_CHANGES);
+    public static final ConfigurationChangeResourceDefinition INSTANCE = new ConfigurationChangeResourceDefinition();
+    public static final String OPERATION_NAME = "list-changes";
+
+    private ConfigurationChangeResourceDefinition() {
+        super(new Parameters(PATH, DomainManagementResolver.getResolver(CORE, MANAGEMENT, SERVICE, CONFIGURATION_CHANGES))
+                .setAddHandler(new ConfigurationChangeResourceAddHandler())
+                .setRemoveHandler(new ConfigurationChangeResourceRemoveHandler()));
+    }
+
+    @Override
+    public void registerOperations(ManagementResourceRegistration resourceRegistration) {
+        super.registerOperations(resourceRegistration);
+        resourceRegistration.registerOperationHandler(ConfigurationChangesHandler.DEFINITION, ConfigurationChangesHandler.INSTANCE);
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        super.registerAttributes(resourceRegistration);
+        resourceRegistration.registerReadWriteAttribute(MAX_HISTORY, null, new MaxHistoryWriteHandler(ConfigurationChangesCollector.INSTANCE));
+    }
+
+    private static class ConfigurationChangeResourceAddHandler extends AbstractAddStepHandler {
+
+        public ConfigurationChangeResourceAddHandler() {
+            super(MAX_HISTORY);
+        }
+
+        @Override
+        protected void populateModel(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+            super.populateModel(context, operation, resource);
+            MaxHistoryWriteHandler.INSTANCE.execute(context, operation);
+        }
+
+    }
+
+    private static class ConfigurationChangeResourceRemoveHandler extends AbstractRemoveStepHandler {
+
+        public ConfigurationChangeResourceRemoveHandler() {
+            ConfigurationChangesCollector.INSTANCE.deactivate();
+        }
+    }
+
+    private static class MaxHistoryWriteHandler implements OperationStepHandler {
+
+        private static final MaxHistoryWriteHandler INSTANCE = new MaxHistoryWriteHandler(ConfigurationChangesCollector.INSTANCE);
+        private final ConfigurationChangesCollector collector;
+
+        private MaxHistoryWriteHandler(ConfigurationChangesCollector collector) {
+            this.collector = collector;
+        }
+
+        @Override
+        public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+            ModelNode maxHistory = MAX_HISTORY.resolveModelAttribute(context, operation);
```

Comment:
This isn't correct for a write-attribute op handler, which has params (name=max-history,value=15), not (max-history=15).